components/post_view/post_attachment_opengraph/post_attachment_opengraph.tsx

```diff
@@ -41,230 +47,212 @@ export type Props = {
-                    imageMetadata={imageMetadata}
-                >
-                    {(safeImageUrl: string) => (
-                        <SizeAwareImage
-                            className='attachment__image attachment__image--opengraph'
-                            src={safeImageUrl}
-                            dimensions={imageMetadata}
-                        />
-                    )}
-                </ExternalImage>
-            </div>
-        );
-    }
+    useEffect(() => setPreviewRemoved(post?.props?.[PostTypes.REMOVE_LINK_PREVIEW] === 'true'), [post]);
 
-    truncateText(text?: string) {
-        const maxLength = 300;
-        const ellipsis = '...';
+    const handleRemovePreview = async (e: React.MouseEvent<HTMLButtonElement>) => {
+        e.preventDefault();
 
-        if (text && text.length > maxLength) {
-            return text.substring(0, maxLength - ellipsis.length) + ellipsis;
-        }
-        return text;
-    }
-
-    handleRemovePreview = async () => {
-        const props = Object.assign({}, this.props.post.props);
+        // prevent the button-click to trigger visiting the link
+        e.stopPropagation();
+        const props = Object.assign({}, post.props);
         props[PostTypes.REMOVE_LINK_PREVIEW] = 'true';
 
         const patchedPost = {
-            id: this.props.post.id,
+            id: post.id,
             props,
         };
 
-        return this.props.actions.editPost(patchedPost);
+        return actions.editPost(patchedPost);
     };
 
-    hasPreviewBeenRemoved() {
-        const {post} = this.props;
-
-        if (!post || !post.props) {
-            return false;
-        }
-
-        return post.props[PostTypes.REMOVE_LINK_PREVIEW] === 'true';
-    }
-
-    render() {
-        if (!this.props.previewEnabled || !this.props.enableLinkPreviews) {
-            return null;
-        }
-
-        if (!this.props.post || isSystemMessage(this.props.post)) {
-            return null;
-        }
-
-        if (this.hasPreviewBeenRemoved()) {
-            return null;
-        }
-
-        const data = this.props.openGraphData;
-        if (!data) {
-            return null;
-        }
-
-        const permalinkClass = this.props.isInPermalink ? 'permalink--opengraph' : '';
+    const toggleImagePreview = (e: React.MouseEvent<HTMLButtonElement>, collapse: boolean) => {
+        e.preventDefault();
 
-        const imageUrl = getBestImageUrl(data, this.props.post.metadata.images);
-        const imageMetadata = this.getImageMetadata(imageUrl);
-        const hasLargeImage = this.isLargeImage(imageMetadata);
+        // prevent the button-click to trigger visiting the link
+        e.stopPropagation();
+        setShowImagePreview(collapse);
+    };
 
-        let removePreviewButton;
-        if (this.props.currentUserId === this.props.post.user_id && !this.props.isInPermalink) {
-            removePreviewButton = (
+    const doNotRender = !rest.previewEnabled ||
```

Comment:
Could you change this back to using one or more early returns like we had before to be more consistent with other code? It'd also avoid the large multi-line ternary statement